css-view-transitions-2/Overview.bs

```diff
@@ -598,20 +615,19 @@ When capturing the old or new state for an element, perform the following steps
 			Note: this means that calling {{Document/startViewTransition()}} while capturing the old document for a cross-document view-transition would run the callback but skip the animation.
 
 		1. Let |outboundTransition| be a new {{ViewTransition}} object in |oldDocument|'s [=relevant Realm=],
-			whose [=ViewTransition/active types=] is |resolvedRule|, and whose [=ViewTransition/process old state captured=] is set to the following steps:
+			whose [=ViewTransition/active types=] is |resolvedRule|.
 
-			Issue: should we check for the opt-in again, in case there was a CSSOM change in a requestAnimationFrame callback?
+		1. Run the following steps when [=Perform pending transition operations|performing pending transition operations=]:
 
-			1. If |outboundTransition|'s [=ViewTransition/phase=] is "`done`", then call |onReady| and return.
+			1. If |outboundTransition|'s [=ViewTransition/phase=] is "`done`", then call |proceedWithNavigation| and return.
 
 			1. Assert: |outboundTransition|'s  [=ViewTransition/phase=] is "`pending-capture`".
 
-			1. [=Clear view transition=] |outboundTransition|.
+			1. Set |oldDocument|'s [=auto-skip view transitions=] to false.
 
-				Note: The ViewTransition object on the old Document should be destroyed after its state has been copied to the new Document below.
-					We explicitly clear it here since the old Document may be cached by the UA.
+			1. [=Capture the old state=] for |outboundTransition|.
 
-			1. Set |oldDocument|'s [=auto-skip view transitions=] to false.
+			1. If failure is returned, call |proceedWithNavigation| and return.
```

Comment:
"proceedWithNavigation" will be invoked from the rendering task source if done from "perform pending transition operations". And whatever task source script runs in if done via `viewTransition.skipTransition()`.

Also, in the case where we don't even set up an outbound transition, "proceedWithNavigation" is invoked synchronously at the html call-site. So now it will run on whatever source "setup cross-document view-transition" is called from.

It seems odd if the navigation steps are running on different task sources in the 3 cases. Can you document the API contract on where navigation steps should be run, based on what happens in the non-VT case. And ensure we consistently use the same task source for the VT case?

Reply:
This is handled in the HTML spec. The callback that's passed here queues steps on the traversal queue (browser process), so it doesn't matter who calls it.